scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -1,62 +1,59 @@
 package com.twitter.scalding.beam_backend
 
+import cascading.flow.FlowDef
 import com.stripe.dagon.Rule
 import com.twitter.scalding.Execution.{ToWrite, Writer}
+import com.twitter.scalding.beam_backend.BeamWriter.tempSources
 import com.twitter.scalding.typed._
-import com.twitter.scalding.{CFuture, CancellationHandler, Config, Execution, ExecutionCounters}
-import java.io.InputStream
+import com.twitter.scalding.{CFuture, CancellationHandler, Config, Execution, ExecutionCounters, Mode}
 import java.nio.channels.Channels
-import java.util.concurrent.atomic.AtomicLong
 import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicLong
 import org.apache.beam.sdk.Pipeline
 import org.apache.beam.sdk.coders.Coder
 import org.apache.beam.sdk.io.FileSystems
 import scala.annotation.tailrec
 import scala.collection.convert.decorateAsScala._
 import scala.concurrent.{ExecutionContext, Future}
 
+case class TempSource[A](path: String, coder: Coder[A]) extends TypedSource[A] {
+  def error = sys.error("beam sources don't work in cascading")
+  def converter[U >: A] = error
+  def read(implicit flowDef: FlowDef, mode: Mode) = error
+}
+
 class BeamWriter(val beamMode: BeamMode) extends Writer {
   private val state = new AtomicLong()
 
   private val sourceCounter: AtomicLong = new AtomicLong(0L)
-  private val iterablePipes: scala.collection.concurrent.Map[TypedPipe[_], (Coder[_], String)] =
-    new ConcurrentHashMap[TypedPipe[_], (Coder[_], String)]().asScala
 
   override def start(): Unit = ()
 
   override def finished(): Unit = ()
```

Comment:
I think you need to iterate through the forcedPipes here and delete them all. If we don't do this, you will probably have problems leaking space.

We had many issues like that in the past.